File: FrontEnd/OrderProcessingFrontEnd/src/Models/PlaceOrder.ts
export interface PlaceOrderItem{
    productName:string,
    quantity:number,
   // priceAtPurchase:number,
}

export interface PlaceOrder{
    items:PlaceOrderItem[],
    paymentMethod:string,
    orderDate:Date,
}

//To get the detail of placed order

export interface GetOrderDetail{
    orderID:number,
     orderDate:Date,
     paymentMethod:string,
     orderStatus:string,
     totalAmount:number,
     items:GetOrderItem[],
     customerName:string

}

export interface GetOrderItem{
productName:string,
quantity:number,
priceAtPurchase:number
}

File: OrderProcessingBackEnd/DTO/PlaceOrder.cs
using Microsoft.AspNetCore.Mvc;
using OrderProcessingBackEnd.Models;
using OrderProcessingBackEnd.Controllers;
using OrderProcessingBackEnd.Data;
using OrderProcessingBackEnd.Repository;
using OrderProcessingBackEnd.DTO;
using OrderProcessingBackEnd.Services;
using Microsoft.AspNetCore.Authorization;
using System.Security.Claims;
using Microsoft.EntityFrameworkCore;

namespace OrderProcessingBackEnd.DTO
{
    public class PlaceOrderItemDTO
    {
        public string ProductName { get; set; }
        public int Quantity { get; set; }
        public decimal PriceAtPurchase { get; set; } // calculated inside backend (not in request body)
        //public int? Stock { get; set; } // optional: backend use only, not in request
    }

    public class PlaceOrderDTO
    {
        public DateTime OrderDate { get; set; }
        public List<PlaceOrderItemDTO> Items { get; set; }
        public string PaymentMethod { get; set; }
    }

    //get the order detail DTO
    public class GetOrderDetailsDTO
    {
        public int OrderID { get; set; }
        public string CustomerName { get; set; }
        public DateTime OrderDate { get; set; }
        public string PaymentMethod { get; set; }
        public string OrderStatus { get; set; }
        public decimal TotalAmount { get; set; }
        public List<GetOrderItemDTO> Items { get; set; }
      //  public int ProductID { get; set; }
    }

    public class GetOrderItemDTO
    {
        public string ProductName { get; set; }
        public int Quantity { get; set; }
        public decimal PriceAtPurchase { get; set; }

    }


}
